src/Console/Command/BeginCommand.php

```diff
@@ -14,18 +18,53 @@ final class BeginCommand extends Command
 {
     private const NAME = 'begin';
 
-    public function __construct()
+    /**
+     * @var \PhpTuf\ComposerStager\Domain\BeginnerInterface
+     */
+    private $beginner;
+
+    public function __construct(BeginnerInterface $beginner)
     {
         parent::__construct(self::NAME);
+        $this->beginner = $beginner;
     }
 
     protected function configure(): void
     {
         $this->setDescription('Begins the staging process by copying the active directory to the staging directory');
     }
 
+    /**
+     * @throws \Symfony\Component\Console\Exception\InvalidArgumentException
+     */
     public function execute(InputInterface $input, OutputInterface $output): int
     {
-        return ExitCode::SUCCESS;
+        /** @var string $activeDir */
+        $activeDir = $input->getOption(Application::ACTIVE_DIR_OPTION);
+        /** @var string $stagingDir */
+        $stagingDir = $input->getOption(Application::STAGING_DIR_OPTION);
+
+        if (!$this->beginner->activeDirectoryExists($activeDir)) {
+            $output->writeln(sprintf('<error>The active directory does not exist at "%s"</error>', $activeDir));
+            return ExitCode::FAILURE;
+        }
+
+        if ($this->beginner->stagingDirectoryExists($stagingDir)) {
+            $output->writeln(sprintf('<error>The staging directory already exists at "%s"</error>', $stagingDir));
```

Comment:
Probably a follow-up, but would it make sense to implement a `--force` option that removes the staging directory beforehand, if it exists?

Reply:
I've taken a note.